charts/redpanda/templates/_helpers.tpl

```diff
@@ -304,6 +304,21 @@ Generate configuration needed for rpk
 {{- dig "mechanism" (include "sasl-mechanism" $) $.user -}}
 {{- end -}}
 
+{{- define "sasl-first-users" -}}
+{{- toJson ( dict "list" .Values.auth.sasl.users ) -}}
+{{- end -}}
+
+{{- define "sasl-user-password" -}}
+{{- $pass := "" -}}
+{{- $user := (first (include "sasl-first-users" .| fromJson).list) -}}
```

Comment:
Why we have conversion from list to json to list? Am I missing something obvious in helm?

Reply:
Why not
```suggestion
{{- $user := (first .Values.auth.sasl.users -}}
```

Reply:
I tried this, i kept getting a {}.auth nil interface error.

Reply:
`{"list": [...]}` is the return string. You're not converting to list (that would be `| list`), you're selecting the "list" key.

Reply:
The key should probably have a better name.